v2/pkg/genruntime/indexer.go

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+type Indexer interface {
+	// Index returns the index of the Indexer. The index can be passed to a registration.Index to
+	// build an index for the controller-runtime client. If Index returns nil, there is nothing to index.
+	// See controller-runtime mgr.GetFieldIndexer().IndexField() for more details.
+	Index() []string
```

Comment:
Because `controller-runtime mgr.GetFieldIndexer().IndexField()` operators on an `IndexerFunc` whose signature is `func(Object) []string`.

The main reason I kept that support for our `Indexer` is twofold:
1. Just to match.
2. If we ever have a `Ref` type that actually refers to a collection of resources, it fits into the existing interface. With this said it would have to be the type itself that referred to a collection, and I'm not sure if we'll ever actually do that in practice.

Happy to restrict this to `string` if you think that's better/clearer.

Reply:
Happy to leave them returning `[]string` - I didn't infer this reasoning by reading the code, which is why I queried.